### PR TITLE
Extend setup page with community methods

### PR DESCRIPTION
**Pull request by mrdrogdrog**
### Component/Part
Documentation

### Description
This PR extends the documentation by a bunch of links to community driven installation methods. It also moves "reverse proxy" to "guides", because it neither a "official installation method" or a "community installation method"

### Steps

- [x] Added / updated documentation
- [x] I read the [contribution documentation](https://github.com/hedgedoc/hedgedoc/blob/develop/CONTRIBUTING.md) and
  made sure that:
  - My commits are signed-off to accept the DCO.
  - This PR targets the correct branch: `master` for 1.x & docs, `develop` for 2.x

### CODE REVIEW

File: docs/mkdocs.yml [hunk 1]
```diff
@@ -8,14 +8,19 @@ edit_uri: https://github.com/hedgedoc/hedgedoc/edit/master/docs/content/
...
-      - 'Reverse Proxy': setup/reverse-proxy.md
-      - Docker: setup/docker.md
-      - Cloudron: setup/cloudron.md
-      - Heroku: setup/heroku.md
-      - LinuxServer: setup/docker-linuxserver.md
-      - Yunohost: setup/yunohost.md
+      - Official installation methods:
+        - Docker: setup/docker.md
+        - Heroku: setup/heroku.md
+        - 'Manual Installation': setup/manual-setup.md
```
Comment (DerMolly) [inline docs/mkdocs.yml:14]: I'd suggest switching Heroku and Manual Installation here

**Review by DerMolly — CHANGES_REQUESTED** [review]

File: docs/mkdocs.yml (hunk 1 above)
Comment (mrdrogdrog) [reply docs/mkdocs.yml]: done

File: docs/content/setup/index.md [hunk 2]
```diff
@@ -1,8 +1,19 @@
...
-- [Cloudron](cloudron)
-- [LinuxServer.io (multi-arch docker)](docker-linuxserver)
 - [Heroku](heroku)
 - [Manual setup](manual-setup)
+
+## Community installation methods
+
+Unfortunately we can't cover every distribution system ourselves, but we can link guides and methods, that are managed by our awesome community:
+
+- [LinuxServer.io (multi-arch docker)](docker-linuxserver)
```
[the remark this answers is not in the document]
Comment (mrdrogdrog) [reply docs/content/setup/index.md:15]: hm. I guess not. If we have every link in the sidebar then it's not necessary.

[the remark this answers is not in the document]
Comment (mrdrogdrog) [reply docs/content/setup/index.md:15]: removed

**Review by DerMolly — APPROVED** [review]

**Review by SISheogorath — APPROVED** [review]

**Review by ErikMichelson — APPROVED** [review]

File: docs/mkdocs.yml [hunk 3]
```diff
@@ -8,14 +8,19 @@ edit_uri: https://github.com/hedgedoc/hedgedoc/edit/master/docs/content/
...
-      - Yunohost: setup/yunohost.md
+      - Official installation methods:
+          - Docker: setup/docker.md
+          - 'Manual Installation': setup/manual-setup.md
+          - Heroku: setup/heroku.md
+      - Community installation methods:
+          - Cloudron: setup/cloudron.md
+          - LinuxServer: setup/docker-linuxserver.md
+          - Yunohost: setup/yunohost.md
+          - 'Helm chart on artifact': https://artifacthub.io/packages/helm/nicholaswilde/hedgedoc
```
Comment (davidmehren) [inline docs/mkdocs.yml:19]: To be consistent with the casing of the other items this should be
```suggestion
          - 'Helm Chart on Artifact Hub': https://artifacthub.io/packages/helm/nicholaswilde/hedgedoc
```

We could also just remove the 'on Artifact Hub' part completely?

File: docs/mkdocs.yml [hunk 4]
```diff
@@ -8,14 +8,19 @@ edit_uri: https://github.com/hedgedoc/hedgedoc/edit/master/docs/content/
...
-      - Cloudron: setup/cloudron.md
-      - Heroku: setup/heroku.md
-      - LinuxServer: setup/docker-linuxserver.md
-      - Yunohost: setup/yunohost.md
+      - Official installation methods:
+          - Docker: setup/docker.md
+          - 'Manual Installation': setup/manual-setup.md
+          - Heroku: setup/heroku.md
+      - Community installation methods:
+          - Cloudron: setup/cloudron.md
```
Comment (davidmehren) [inline docs/mkdocs.yml:16]:
```suggestion
          - 'Cloudron App': setup/cloudron.md
```

File: docs/mkdocs.yml [hunk 5]
```diff
@@ -8,14 +8,19 @@ edit_uri: https://github.com/hedgedoc/hedgedoc/edit/master/docs/content/
...
+          - LinuxServer: setup/docker-linuxserver.md
```
Comment (davidmehren) [inline docs/mkdocs.yml:17]:
```suggestion
          - 'LinuxServer Docker': setup/docker-linuxserver.md
```
This sounds a bit weird, so we might add 'Image' both here and for our own image

File: docs/mkdocs.yml [hunk 6]
```diff
@@ -8,14 +8,19 @@ edit_uri: https://github.com/hedgedoc/hedgedoc/edit/master/docs/content/
...
+          - Yunohost: setup/yunohost.md
+          - 'Helm chart on artifact': https://artifacthub.io/packages/helm/nicholaswilde/hedgedoc
+          - 'Arch Linux': https://archlinux.org/packages/community/any/hedgedoc/
```
Comment (davidmehren) [inline docs/mkdocs.yml:20]:
```suggestion
          - 'Arch Linux Package': https://archlinux.org/packages/community/any/hedgedoc/
```

File: docs/mkdocs.yml [hunk 7]
```diff
@@ -8,14 +8,19 @@ edit_uri: https://github.com/hedgedoc/hedgedoc/edit/master/docs/content/
...
-      - LinuxServer: setup/docker-linuxserver.md
-      - Yunohost: setup/yunohost.md
+      - Official installation methods:
+          - Docker: setup/docker.md
+          - 'Manual Installation': setup/manual-setup.md
+          - Heroku: setup/heroku.md
+      - Community installation methods:
+          - Cloudron: setup/cloudron.md
+          - LinuxServer: setup/docker-linuxserver.md
+          - Yunohost: setup/yunohost.md
```
Comment (davidmehren) [inline docs/mkdocs.yml:18]:
```suggestion
          - 'YunoHost App' : setup/yunohost.md
```

**Review by davidmehren — CHANGES_REQUESTED** [review]
I like the split, but find it a bit weird to have external links in the left navbar. Maybe we can find a way around them or mark them somehow?

I also added some suggestions that hopefully make a bit clearer what the community-methods are.

File: docs/mkdocs.yml (hunk 3 above)
Comment (mrdrogdrog) [reply docs/mkdocs.yml]: Done

File: docs/mkdocs.yml (hunk 4 above)
Comment (mrdrogdrog) [reply docs/mkdocs.yml]: Done

File: docs/mkdocs.yml (hunk 5 above)
Comment (mrdrogdrog) [reply docs/mkdocs.yml]: Done

File: docs/mkdocs.yml (hunk 6 above)
Comment (mrdrogdrog) [reply docs/mkdocs.yml]: Done

File: docs/mkdocs.yml (hunk 7 above)
Comment (mrdrogdrog) [reply docs/mkdocs.yml]: Done

**Comment by mrdrogdrog** [comment]
@davidmehren  I created subpages for archlinux and helm, because I can't mark external links.

@DerMolly @ErikMichelson @SISheogorath please let me know if you're against this change.

**Review by InnayTool — APPROVED** [review]

**Review by davidmehren — APPROVED** [review]

**Pull request merged**